src/pke/lib/scheme/ckksrns/ckksrns-fhe.cpp

```diff
@@ -61,6 +61,26 @@ void FHECKKSRNS::EvalBootstrapSetup(const CryptoContextImpl<DCRTPoly>& cc, std::
     uint32_t M     = cc.GetCyclotomicOrder();
     uint32_t slots = (numSlots == 0) ? M / 4 : numSlots;
 
+    // Set correction factor by default, if it is not already set.
+    if (correctionFactor == 0.0) {
+        if (cryptoParams->GetScalingTechnique() == FLEXIBLEAUTOEXT) {
```

Comment:
Please check if we can add FLEXIBLEAUTO here.

Reply:
Ran all the experiments for FLEXIBLEAUTO, and they give exactly the same best values for the correction factor as FLEXIBLEAUTOEXT. We also see similar values for precision though (20 bits for full packing with ring dimension 2^16), so I'm not sure if this makes sense.

Reply:
I examined this further and ran experiments for `FLEXIBLEAUTOEXT`, `FLEXIBLEAUTO`, and `FIXEDAUTO`. You are correct that there is no difference in the bootstrapping precision between `FLEXIBLEAUTOEXT` and `FLEXIBLEAUTO`. In this case, the deterministic error due to the approximation $sin(x) \approx x$ is much higher than the nondeterministic error due to RLWE noise. All `FLEXIBLEAUTOEXT` does is reduces the fresh encryption error to the rescaling error, but the accumulated nondeterministic error turns out to be lower than then the deterministic sine wave approximation error incurred in bootstrapping.

Note that `FIXEDAUTO` has a significantly lower precision.